carbon-design-system/ibm-security
# fix(delimited-list): update component when element.current or items change

## Affected issues

- Resolves #371 

## Proposed changes

- refactor to use hooks
- `useLayoutEffect` to update `hasOverflow` if `element.current` or `items` change

## Testing instructions

- in the storybook environment, play around with adding/removing `items` in the knobs, and also toggling `truncate`. Keep in mind that the component right now is set to `width: 20rem` -- so as you add/remove `items` from the array, expect the "item counter" to be there depending on if you change the `items` to be shorter (`scrollWidth`) and the visible list container (`clientWidth`).
- Also keep in mind that the `truncate` prop can change things... so even if you have a LOT of items, setting `truncate={false}` will take away the item counter regardless of what the `hasOverflow` state is.
